deploy/batch-rca-automation/batch_rca_headless.sh

```
@@ -295,42 +274,91 @@ if grep -q "MLFLOW_CLAUDE_TRACING_ENABLED.*true" "$SETTINGS_FILE" 2>/dev/null; t
+DURATION=$(echo "$CLAUDE_OUTPUT" | jq -r '.duration_ms // "unknown"')
+NUM_TURNS=$(echo "$CLAUDE_OUTPUT" | jq -r '.num_turns // "unknown"')
+echo "[METRICS] cost=\$${COST} duration=${DURATION}ms turns=${NUM_TURNS}"
+
+RESULT_RAW=$(echo "$CLAUDE_OUTPUT" | jq -r '.result // empty' 2>/dev/null)
+
+# Try parsing .result directly as JSON
+if echo "$RESULT_RAW" | jq -e . >/dev/null 2>&1; then
+  echo "$RESULT_RAW" > "$REPORT_FILE"
+  echo "[INFO] Report written to $REPORT_FILE"
+# Strip markdown code fences and try again
+elif RESULT_STRIPPED=$(echo "$RESULT_RAW" | sed -n '/^```json\s*$/,/^```\s*$/{ /^```/d; p; }' | jq -e . 2>/dev/null); then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)